examples/catalog/lib/custom_a11y_traversal.dart

```diff
@@ -0,0 +1,342 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/semantics.dart';
+
+/// This example shows a set of widgets for changing data fields arranged in a column
+/// of rows but, in accessibility mode, are traversed in a custom order.
+///
+/// This demonstrates how Flutter's accessibility system describes custom traversal
+/// orders using sort keys.
+///
+/// The example app here has three fields that have a title and up/down spinner buttons
+/// above and below. The traversal order should allow the user to focus on each
+/// title, the input field next, the up spinner next, and the down spinner last before
+/// moving to the next input title.
+///
+/// Users that do not use a screen reader will just see a regular app with controls.
+///
+/// The example's [RowColTraversal] widget sets up two [Semantics] objects that wrap the
+/// given [Widget] child, providing the traversal order they should have in the "row"
+/// direction, and then the traversal order they should have in the "column" direction.
+///
+/// Sort keys, by default, are appended to the sort orders for their parents, but
+/// they can also override those of their parents (by setting
+/// [SemanticsSortOrder.discardParentOrder] to true), and an entire sort order can be
+/// defined with multiple keys, to provide for virtually any ordering.
+///
+/// Keys at the same position in the sort order are compared with each other, and
+/// keys which are of different types, or which have different [SemanticSortKey.name]
+/// values compare as "equal" so that two different types of keys can co-exist at the
+/// same level and not interfere with each other, allowing for sorting in groups.
+/// Keys that evaluate as equal, or when compared with Widgets that don't have
+/// [Semantics], are given the default upper-start-to-lower-end geometric
```

Comment:
I believe its currently top-left to bottom-right no matter of direction? 
https://github.com/flutter/engine/pull/4540/files#diff-be56efcdd49538955ac6ed238eda9ed2R447